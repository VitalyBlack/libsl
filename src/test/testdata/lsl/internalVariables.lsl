libsl "1.0.0";
library simple;

types {
   Int(int);
}

automaton A {
   var b: Int = new B(state = s);

   fun foo() {
       b.i = 1;
   }
}

automaton B {
   state s;
   var i: Int;
}