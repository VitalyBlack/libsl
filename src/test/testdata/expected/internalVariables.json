{
  "name": "simple",
  "lslVersion": "1.0.0",
  "types": [
    {
      "name": "Int",
      "realName": "int"
    }
  ],
  "automata": [
    {
      "name": "A",
      "constructorVariables": [],
      "variables": [
        {
          "name": "i",
          "type": "Int"
        },
        {
          "name": "b",
          "type": "Int"
        }
      ],
      "states": [],
      "shifts": [],
      "functions": [
        {
          "name": "foo",
          "automaton": "A",
          "args": [],
          "contracts": [],
          "statements": [
            {
              "kind": "assignment",
              "variableName": "b.i",
              "variableAutomaton": "A",
              "value": {
                "kind": "integer",
                "value": 1
              }
            }
          ]
        }
      ]
    },
    {
      "name": "B",
      "constructorVariables": [],
      "variables": [
        {
          "name": "i",
          "type": "Int"
        }
      ],
      "states": [
        {
          "name": "s",
          "kind": "SIMPLE"
        }
      ],
      "shifts": [],
      "functions": []
    }
  ]
}